### PR TITLE
Complete ReWork of the openHab handling in the integration 

### PR DESCRIPTION
**Pull request by miloit**
Update of online check of openhab system by using http status code
Json Document Handling improvement
Code cleaning
Bugfix the openHab Event Server Handling
Bugfix of the JSON Document handling

### CODE REVIEW

File: src/openhab.h [hunk 1]
```diff
@@ -99,18 +100,30 @@ class OpenHAB : public Integration {
...
+    bool _flagOpenHabConnected = false;
+    // QObject* context = new QObject(this);
+    QString        _tempJSONData = "";
+    OpenHAB*       context_openHab;
+    bool           _flagSseConnected = false;
+    bool           _flagMoreDataNeeded = false;
+    QJsonDocument  doc;
+    QJsonDocument  pyload;
+    QByteArray     rawData;
+    QByteArrayList splitted;
```
Comment (zehnm) [inline src/openhab.h]: These variables seem to be used only locally and don't need to be class variables. See comments in cpp file.

File: src/openhab.h [hunk 2]
```diff
@@ -99,18 +100,30 @@ class OpenHAB : public Integration {
...
+    QTimer*                 _sseReconnectTimer;
+    QString                 _url;
+    QString                 _token;
+    int                     _networktries = 0;
+    QNetworkAccessManager*  _nam;
+    bool                    _flagleaveStandby = false;
+    QList<EntityInterface*> _myEntities;  // Entities of this integration
+    QRegExp                 regex_colorvalue =
+        QRegExp("[0-9]?[0-9]?[0-9][,][0-9]?[0-9]?[0-9][,][0-9]?[0-9][.]?[0-9]?[0-9]?[0-9]?[0-9]?");
     QRegExp regex_brightnessvalue = QRegExp("[1]?[0-9]?[0-9]");
```
Comment (zehnm) [inline src/openhab.h]: Please stick to one naming convention: these variables have no prefix as the others and are not lowerCamelCase

File: src/openhab.h [hunk 3]
```diff
@@ -99,18 +100,30 @@ class OpenHAB : public Integration {
...
-    bool                        _flagStandby;
-    bool                        _flagOpenHabConnected = false;
-    QObject* context = new QObject(this);
+    int     _tries;
+    bool    _flagStandby;
+    // bool     _flagprocessitems = false;
+    bool _flagOpenHabConnected = false;
+    // QObject* context = new QObject(this);
+    QString        _tempJSONData = "";
+    OpenHAB*       context_openHab;
```
Comment (zehnm) [inline src/openhab.h:122]: Please stick to one naming convention: variable has no prefix as the others and no lowerCamelCase

File: src/openhab.cpp [hunk 4]
```diff
@@ -70,100 +66,161 @@ OpenHAB::OpenHAB(const QVariantMap& config, EntitiesInterface* entities, Notific
...
-    _sseReconnectTimer = new QTimer(this);
-
-
-
-    qCDebug(m_logCategory) << "setup";
+    context_openHab = this;
+    _sseNetworkManager = new QNetworkAccessManager(context_openHab);
+    _sseReconnectTimer = new QTimer(context_openHab);
+    _nam = new QNetworkAccessManager(context_openHab);
+    _nam = new QNetworkAccessManager(context_openHab);
```
Comment (zehnm) [inline src/openhab.cpp]: `_nam` doesn't need to be allocated twice.
There's also no need for dynamic allocation.

File: src/openhab.cpp [hunk 5]
```diff
@@ -70,100 +66,161 @@ OpenHAB::OpenHAB(const QVariantMap& config, EntitiesInterface* entities, Notific
...
+    if (_sseReply->error() == QNetworkReply::NoError) {
+        QJsonParseError parseerror;
 
-    for (QByteArray data : splitted) {
-        if ((data.length() == 0) || (data.startsWith("event: message"))) {
-            continue;
-        }
+        rawData.clear();
+        rawData = _sseReply->readAll();
+        splitted = rawData.split('\n');
```
Comment (zehnm) [inline src/openhab.cpp:94]: Use a local variables instead of class variables. They are only used inside this method.

File: src/openhab.cpp [hunk 6]
```diff
@@ -70,100 +66,161 @@ OpenHAB::OpenHAB(const QVariantMap& config, EntitiesInterface* entities, Notific
...
+            // only process state changes
+            if (!_flagMoreDataNeeded) {
+                if ((doc.object().value("type").toString() == "ItemStateEvent") ||
+                    (doc.object().value("type").toString() == "GroupItemStateChangedEvent")) {
+                    // get item name from the topic string
+                    // example: smarthome/items/EG_Esszimmer_Sonos_CurrentPlayingTime/state
+                    QString          name = doc.object().value("topic").toString().split('/')[2];
+                    EntityInterface* entity = m_entities->getEntityInterface(name);
+                    if (entity != nullptr && entity->connected()) {
+                        pyload =
```
Comment (zehnm) [inline src/openhab.cpp:139]: Use a local variable for pyoad. Doesn't seem to be used outside this method.

File: src/openhab.cpp [hunk 7]
```diff
@@ -70,100 +66,161 @@ OpenHAB::OpenHAB(const QVariantMap& config, EntitiesInterface* entities, Notific
...
-        QJsonDocument doc = QJsonDocument::fromJson(data.mid(6), &parseerror);
-        if (parseerror.error != QJsonParseError::NoError) {
-            // qCDebug(m_logCategory) << QString(doc.toJson(QJsonDocument::Compact));
-            // qCDebug(m_logCategory) << "read " << rawData.size() << "bytes";
-            qCCritical(m_logCategory) << "SSE JSON error:" << parseerror.errorString();
-            continue;
-        }
+            if (_flagMoreDataNeeded) {
+                _tempJSONData = _tempJSONData + QString(data);
+                doc = QJsonDocument::fromJson(_tempJSONData.toUtf8(), &parseerror);
```
Comment (zehnm) [inline src/openhab.cpp:103]: Use a local variable for doc. Doesn't seem to be used outside this method.

File: src/openhab.cpp [hunk 8]
```diff
@@ -173,185 +230,212 @@ void OpenHAB::startSse() {
...
+                true, tr("Cannot connect to ").append(friendlyName()).append("."), tr("Reconnect"),
+                [](QObject* param) {
+                    Integration* i = qobject_cast<Integration*>(param);
+                    i->connect();
+                },
+                this);
+            _flagOpenHabConnected = false;
+            disconnect();
+            qCDebug(m_logCategory) << "disconnect 3";
+            qCDebug(m_logCategory) << "openhab not reachable";
```
Comment (zehnm) [inline src/openhab.cpp]: Merge log statements

**Review by zehnm — CHANGES_REQUESTED** [review]

File: src/openhab.h (hunk 1 above)
Comment (miloit) [reply src/openhab.h]: done

File: src/openhab.h (hunk 2 above)
Comment (miloit) [reply src/openhab.h]: done

File: src/openhab.cpp (hunk 6 above)
Comment (miloit) [reply src/openhab.cpp:139]: done

File: src/openhab.cpp (hunk 7 above)
Comment (miloit) [reply src/openhab.cpp:103]: done

File: src/openhab.cpp (hunk 8 above)
Comment (miloit) [reply src/openhab.cpp]: done

File: src/openhab.cpp (hunk 4 above)
Comment (miloit) [reply src/openhab.cpp]: done

**Review by zehnm — APPROVED** [review]

**Pull request merged**